website/docs/basics.md

```diff
@@ -0,0 +1,214 @@
+  The logical name establishes a notion of identity within Pulumi, and the physical name is used as identity by the provider
+- a [resource type](https://www.pulumi.com/docs/concepts/resources/names/#types), which identifies the provider and the kind of resource being created
+- [URN](https://www.pulumi.com/docs/concepts/resources/names/#types), which is an automatically constructed globally unique identifier for the resource.
+
+Each resource also can have:
+- a set of [arguments](https://www.pulumi.com/docs/concepts/resources/properties/) that define the behavior of the resulting infrastructure
+- and a set of [options](https://www.pulumi.com/docs/concepts/options/) that control how the resource is created and managed by the Pulumi engine.
+
+Every is managed by a [provider](https://www.pulumi.com/docs/concepts/resources/providers/) which is a plugin that
+provides the implementation details. If not specified explicitly, the default provider is used.
+Providers can be configured using [provider configuration](https://www.pulumi.com/docs/concepts/resources/providers/#explicit-provider-configuration).
+
+#### Inputs and Outputs
+
+Inputs and Outputs are the primary [asynchronous data types in Pulumi](https://www.pulumi.com/docs/concepts/inputs-outputs/), 
+and they signify values that will be provided by the engine later, when the resource is created and its properties can be fetched.
+`Input[A]` type is an alias for `Output[A]` type used by [resource](#resources) arguments.
+
+Outputs are values of type `Output[A]` and behave very much like [monads](https://en.wikipedia.org/wiki/Monad_(functional_programming)). 
+This is necessary because output values are not fully known until the infrastructure resource has actually completed 
+provisioning, which happens asynchronously after the program has finished executing.
+
+Outputs are used to:
+- automatically captures dependencies between [resources](#resources)
+- provide a way to express transformations on its value before it's known
+- deffer the evaluation of its value until it's known
+- track the _secretness_ of its value
+
+Output transformations available in Besom:
+- [`map` and `flatMap`](apply_methods) methods take a callback that receives the plain value, and computes a new output
+- [lifting](lifting) directly read properties off an output value
+- [interpolation](interpolator) concatenate string outputs with other strings directly
+
+To create an output from a plain value, use the `Output` constructor, e.g.:
+```scala
+val hello = Output("hello")
+```
+
+If you have multiple outputs and need to use them together you can use the standard 
+[`zip`](https://scala-lang.org/api/3.x/scala/collection/View.html#zip-1dd) 
+method to combine them into a single output:
+
+```scala
+val port: Output[Int] = pod.port
+val host: Output[String] = node.hostname
+val hello = host.zip(port).map((a, b) => s"https://$hostname:$port/")
```

Comment:
not 100% sure about the syntax

Reply:
it looks correct
maybe we should consider using mdoc to type-check the snippets?

Reply:
Checking snippets sounds great! Yes, there was a typo.

This compiles:
```
  val port: Output[Int] = Output(8080)
  val host: Output[String] = Output("localhost")
  val hello = host.zip(port).map((hostname, port) => s"https://$hostname:$port/")
```

I've also improved the snippet a bit.

Reply:
I've change the snipped to be more clear:
```
val port: Output[Int] = pod.port
val host: Output[String] = node.hostname
val hello: Output[(String, Int)] = host.zip(port)
val url: Output[String] = hello.map { case (a, b) => s"https://$hostname:$port/" }
```